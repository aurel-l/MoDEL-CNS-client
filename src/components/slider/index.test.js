import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';

import Slider from '.';

describe('<Slider />', () => {
  let wrapper;

  afterAll(cleanup);

  describe('no projection selected', () => {
    afterAll(() => wrapper.unmount());

    it('should render', () => {
      wrapper = render(<Slider value={50} label="Some label:" />);
      expect(wrapper.container).toBeInstanceOf(HTMLElement);
      expect(wrapper.container).toMatchSnapshot();
    });

    it('should render without label', () => {
      wrapper = render(<Slider value={50} className="button-class" />);
      expect(wrapper.container).toBeInstanceOf(HTMLElement);
      expect(wrapper.container).toMatchSnapshot();
    });

    it('should handle clicks', () => {
      expect(
        wrapper.container.parentElement.querySelector('[data-popover]'),
      ).toBeNull();

      // open
      fireEvent.click(wrapper.container.querySelector('.button-class'));
      const popover = wrapper.container.parentElement.querySelector(
        '[data-popover] > [role=document]',
      );
      expect(popover).toBeInstanceOf(HTMLElement);

      // close by clicking on backdrop
      fireEvent.click(
        wrapper.container.parentElement.querySelector(
          '[data-popover] > :not([role=document])',
        ),
      );
      expect(
        wrapper.container.parentElement.querySelector(
          '[data-popover]:not([aria-hidden=true]) > [role=document]',
        ),
      ).toBeNull();
    });

    it('should handle click to open, escape to close', () => {
      // open
      fireEvent.click(wrapper.container.querySelector('.button-class'));
      const popover = wrapper.container.parentElement.querySelector(
        '[data-popover] > [role=document]',
      );
      expect(popover).toBeInstanceOf(HTMLElement);

      // close by escaping
      fireEvent.keyDown(popover, { key: 'Escape', code: 27 });
      expect(
        wrapper.container.parentElement.querySelector(
          '[data-popover]:not([aria-hidden=true]) > [role=document]',
        ),
      ).toBeNull();
    });
  });
});
