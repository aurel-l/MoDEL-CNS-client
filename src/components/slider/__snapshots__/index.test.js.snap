// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> no projection selected should render 1`] = `
<div>
  <button
    class="MuiButtonBase-root-7 MuiIconButton-root-1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label-6"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-10"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15 21h2v-2h-2v2zm4-12h2V7h-2v2zM3 5v14c0 1.1.9 2 2 2h4v-2H5V5h4V3H5c-1.1 0-2 .9-2 2zm16-2v2h2c0-1.1-.9-2-2-2zm-8 20h2V1h-2v22zm8-6h2v-2h-2v2zM15 5h2V3h-2v2zm4 8h2v-2h-2v2zm0 8c1.1 0 2-.9 2-2h-2v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root-22"
    />
  </button>
</div>
`;

exports[`<Slider /> no projection selected should render without label 1`] = `
<div>
  <button
    class="MuiButtonBase-root-7 MuiIconButton-root-1 button-class"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label-6"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-10"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15 21h2v-2h-2v2zm4-12h2V7h-2v2zM3 5v14c0 1.1.9 2 2 2h4v-2H5V5h4V3H5c-1.1 0-2 .9-2 2zm16-2v2h2c0-1.1-.9-2-2-2zm-8 20h2V1h-2v22zm8-6h2v-2h-2v2zM15 5h2V3h-2v2zm4 8h2v-2h-2v2zm0 8c1.1 0 2-.9 2-2h-2v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root-22"
    />
  </button>
</div>
`;
